src/mpags-cipher.cpp

```diff
@@ -88,23 +96,18 @@ int main(int argc, char* argv[])
             inputText += transformChar(inputChar);
         }
     }
-
-    std::string outputText;
-
-    switch (settings.cipherType[0]) {
-        case CipherType::Caesar: {
-            // Run the Caesar cipher (using the specified key and encrypt/decrypt flag) on the input text
-            CaesarCipher cipher{settings.cipherKey[0]};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
-        }
-        case CipherType::Playfair: {
-            PlayfairCipher cipher{settings.cipherKey[0]};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
-        }
+    
+     std::vector<std::unique_ptr<Cipher>> ciphers;     // Store cipher instances
```

Comment:
Can reserve the memory for `settings.nExpectedCiphers`:
```suggestion
     std::vector<std::unique_ptr<Cipher>> ciphers;     // Store cipher instances
     ciphers.reserve( settings.nExpectedCiphers );
```